AndroidTVGateway/agent/app/src/main/java/org/wso2/androidtv/agent/siddhiSinks/EdgeGatewaySink.java

```diff
@@ -148,45 +179,92 @@ public void publish(Object o, DynamicOptions dynamicOptions) throws ConnectionUn
             try {
                 jsonEvent.put("payloadData", payload);
             } catch (JSONException e) {
-                Log.e("EdgeGatewaySink","Error while inserting values to JSON object");
+                Log.e(TAG_SINK,"Error while inserting values to JSON object", e);
             }
 
             JSONObject wrapper = new JSONObject();
             wrapper.put("event", jsonEvent);
 
             if (androidTVMQTTHandler != null) {
+
+                H2Connection H2Conn = new H2Connection();
+
                 if (androidTVMQTTHandler.isConnected()) {
+
+                    if(hasPersistedData){
```

Comment:
Please format the code according to WSO2 coding styles.
there will be a space between if and (